waspc/data/Generator/templates/react-app/src/operations/resources.js

```diff
@@ -45,3 +45,22 @@ export const invalidateQueriesUsing = (resources) => {
     queryClient.invalidateQueries(queryCacheKey)
   }
 }
+
+export function removeQueries() {
+  // TODO(filip): We are currently removing all the queries, but we should
+  // remove only non-public, user-dependent queries - public queries are
+  // expected not to change in respect to the currently logged in user.
+  queryClient.removeQueries()
+}
+
+// TODO(filip): We are currently invalidating and removing  all the queries, but
+// we should remove only the non-public, user-dependent ones.
+export function invalidateAndClearQueries() {
+  // If we don't invalidate the queries before removing them, Wasp will stay on
```

Comment:
Is it only about not staying on the same page? Because that doesn't sound like a strong enough reason to do this. There are other ways to move user to another page, more explicit ones.

Reply:
That is definitely, I've added a TODO for making an issue that addresses this. I didn't want to change too much at the moment to avoid prolonging the PR.

Forgot to mention it, sorry!

Reply:
Anyway, I wouldn't make any changes here in this PR - that's how it's been working until now. I only added a comment to simplify refactoring it later.

Reply:
Yup sounds good!